waku/peer.go

```diff
@@ -277,6 +315,29 @@ func (p *Peer) bloomMatch(env *Envelope) bool {
 	return p.fullNode || BloomFilterMatch(p.bloomFilter, env.Bloom())
 }
 
+func (p *Peer) topicInterestMatch(env *Envelope) bool {
+	p.topicInterestMu.Lock()
+	defer p.topicInterestMu.Unlock()
+
+	if p.topicInterest == nil {
+		return false
+	}
+
+	return p.fullNode || p.topicInterest[env.Topic]
+}
+
+func (p *Peer) topicOrBloomMatch(env *Envelope) bool {
```

Comment:
I would explain here or somewhere else how we treat `len(topicInterest) == 0` and zero bloom filter. It is not obvious looking at the code.